src/aiida/tools/pytest_fixtures/configuration.py

```diff
@@ -0,0 +1,243 @@
+
+            This ensures that the contents of the profile are reset as well as the ``Manager``, which may hold
+            references to data that will be destroyed. The daemon will also be stopped if it was running.
+            """
+            from aiida.engine.daemon.client import DaemonException, get_daemon_client
+            from aiida.orm import User
+
+            if broker_backend:
+                daemon_client = get_daemon_client()
+
+                if daemon_client.is_daemon_running:
+                    try:
+                        daemon_client.stop_daemon(wait=True)
+                    except DaemonException:
+                        pass
+
+            manager.get_profile_storage()._clear()
+            manager.reset_profile()
+
+            User(email=profile.default_user_email or email).store()
+
+        # Add the ``reset_storage`` method, such that users can empty the storage through the ``Profile`` instance that
+        # is returned by this fixture.
+        setattr(profile, 'reset_storage', reset_storage)
+
+        with profile_context(profile, allow_switch=True):
+            yield profile
+
+    return factory
+
+
+@pytest.fixture(scope='session', autouse=True)
+def aiida_config(tmp_path_factory, aiida_config_factory):
+    """Return a loaded temporary AiiDA configuration directory.
+
+    This fixture is session-scoped and used automatically as soon as these fixtures are imported.
+
+    :returns :class:`~aiida.manage.configuration.config.Config`: The loaded temporary config.
+    """
+    with aiida_config_factory(tmp_path_factory.mktemp(secrets.token_hex(16))) as config:
+        yield config
+
+
+@pytest.fixture(scope='session', autouse=True)
+def aiida_profile(aiida_config, aiida_profile_factory):
+    """Return a loaded temporary AiiDA profile.
+
+    This fixture is session-scoped and used automatically as soon as these fixtures are imported. The profile defines
+    no broker and uses the ``core.sqlite_dos`` storage backend, meaning it requires no services to run.
+
+    :returns :class:`~aiida.manage.configuration.profile.Profile`: The loaded temporary profile.
+    """
+    with aiida_profile_factory(aiida_config) as profile:
+        yield profile
+
+
+@pytest.fixture(scope='function')
+def aiida_profile_clean(aiida_profile):
+    """Return a loaded temporary AiiDA profile.
+
+    This is a function-scoped version of the ``aiida_profile`` fixture.
+
+    :returns :class:`~aiida.manage.configuration.profile.Profile`: The loaded temporary profile.
+    """
+    aiida_profile.reset_storage()
+    yield aiida_profile
+
+
+@pytest.fixture(scope='class')
+def aiida_profile_clean_class(aiida_profile):
+    """Return a loaded temporary AiiDA profile.
+
+    This is a class-scoped version of the ``aiida_profile`` fixture.
+
+    :returns `~aiida.manage.configuration.profile.Profile`: The loaded temporary profile.
+    """
+    aiida_profile.reset_storage()
+    yield aiida_profile
```

Comment:
I wonder about the semantics here: should `reset_storage` be called at teardown as well?
In any case, I think given that `aiida_profile` is session-scoped, whatever the behaviour is (both at the start and after the test) it should be explicitly documented. (same for `aiida_profile_clean`)

Reply:
I don't think it should also be called at teardown. The `aiida_profile` just provides a profile with initialized storage, it makes no guarantees about its contents. If the test requires an emtpy storage, then the test writer can make use of `aiida_profile_clean` but the onus of having a clean storage lies with the test itself. So I don't think it is necessary to always clean, because then you would really have to make `aiida_profile` function scoped and always call `reset_storage` but that is unnecessarily costly I think.

Reply:
Sounds good, thanks!